### PR TITLE
Fix enabling of Use Serial Mesh

### PR DESCRIPTION
**Pull request by mperego**
Check whether parallel ExodusII is available and use it to conditionally enable Use Serial Mesh. Thanks @bartgol for helping.

### CODE REVIEW

**Comment by mperego** [comment]
This is related to #761.

**Comment by ikalash** [comment]
Thanks @mperego !  I'll test this out on camobap later this afternoon.

**Comment by mperego** [comment]
@ikalash Thanks!
B.t.w. It appears that a couple of tests

> CahnHillElast2D_Epetra
> CahnHillElast2D_Tempus

were if-guarded by Albany_IOPX and were never enabled. This PR enables them and they are now failing:

>  Error, the parameter {name="Initial Order",type="int",value="0"}
> in the parameter (sub)list "Albany Parameters->Piro->Tempus->Tempus Integrator->Time Step Control"
> was not found in the list of valid parameters!

**Review by ikalash — CHANGES_REQUESTED** [review]
Could you please add a print of whether ALBANY_PARALLEL_EXODUS is true/false in the CMakeLists.txt file?  Would be useful to see this in the configure step, for debugging, etc.

**Comment by ikalash** [comment]
> @ikalash Thanks! B.t.w. It appears that a couple of tests
> 
> > CahnHillElast2D_Epetra
> > CahnHillElast2D_Tempus
> 
> were if-guarded by Albany_IOPX and were never enabled. This PR enables them and they are now failing:
> 
> > Error, the parameter {name="Initial Order",type="int",value="0"}
> > in the parameter (sub)list "Albany Parameters->Piro->Tempus->Tempus Integrator->Time Step Control"
> > was not found in the list of valid parameters!

Thanks for the heads up - I'll fix this.

File: CMakeLists.txt [hunk 1]
```diff
@@ -188,6 +188,27 @@ if (SET_COMPILERS_AUTOMATICALLY)
...
+set(CMAKE_REQUIRED_INCLUDES  ${Trilinos_INCLUDE_DIRS})
+check_c_source_compiles(
+  "
+  #include <exodusII_par.h>
+  int main()
+  {
+    return 0;
+  }
+  "
+  PARALLEL_EXODUS_SUPPORTED
```
Comment (bartgol) [inline CMakeLists.txt:202]: Is there a reason not to just use `ALBANY_PARALLEL_EXODUS` as output variable?

**Review by bartgol — APPROVED** [review]
Looks good. I have one comment, but not very important.

File: CMakeLists.txt (hunk 1 above)
Comment (mperego) [reply CMakeLists.txt:202]: Not really. I thought that  
`Performing Test PARALLEL_EXODUS_SUPPORTED - Success`
was more expressive but at the same time wanted to keep Albany way of naming variables. I can change that.

**Comment by mperego** [comment]
> Could you please add a print of whether ALBANY_PARALLEL_EXODUS is true/false in the CMakeLists.txt file? Would be useful to see this in the configure step, for debugging, etc.

Yes, what message would you like to print? I don't see it done in a consistent way. Something like
`ALBANY_PARALLEL_EXODUS set to True` ?

File: CMakeLists.txt (hunk 1 above)
Comment (bartgol) [reply CMakeLists.txt:202]: No, it's fine. I was just curious if there was some weird techincality in the CheckFunctionExists macro maybe. This is fine.

**Comment by ikalash** [comment]
> > Could you please add a print of whether ALBANY_PARALLEL_EXODUS is true/false in the CMakeLists.txt file? Would be useful to see this in the configure step, for debugging, etc.
> 
> Yes, what message would you like to print? I don't see it done in a consistent way. Something like `ALBANY_PARALLEL_EXODUS set to True` ?

Yes exactly - to be consistent with the other messages printed when you configure Albany.

**Review by bartgol — APPROVED** [review]

**Comment by ikalash** [comment]
This looks good.  I just verified that everything is OK on camobap with your change.  I'll go ahead and merge.

**Comment by mperego** [comment]
@ikalash thanks for checking camobap. I usually prefer to rebase and merge to avoid useless merge commits.

B.t.w. the failing CahnHill tests don't appear on Camobap because ALBANY_PARALLEL_EXODUS is false there.

**Comment by ikalash** [comment]
@mperego , yes, thank you for the reminder.  I'll fix it in master.

**Pull request merged**